src/module.js

```diff
@@ -46,6 +47,16 @@ function module_variable(observer) {
   return new Variable(TYPE_NORMAL, this, observer);
 }
 
+function module_evaluate(name) {
+  var v = this._scope.get(name);
+  if (!v) throw new RuntimeError(name + " is not defined");
+  if (v._observer === no_observer) {
+    v._observer = true;
+    this._runtime._dirty.add(v);
```

Comment:
Should `evaluate` not unconditionally mark the variable in question as `dirty`, before calling `runtime._compute()`?

Imagine a variable that reads from an impure value, like Date.now(). The name `module.evaluate()` makes me think that the named variable is going to be re-computed, and if it isn't — I'll get unexpected results.

Reply:
*runtime*._dirty is just for reachability changes—whether or not a variable is “visible” (and should be computed). We don’t want to recompute reachability if the variable already has an observer.

But, if we do want *module*.evaluate to *recompute* a variable, then we should add it to *runtime*._updates unconditionally, as *variable*.define does. Personally… I’m not so sure that recomputing the variable is what you want, since it means that *module*.evaluate now has a side-effect similar to Shift-Enter on the variable you are evaluating. Would it be better if I renamed this method to *module*.value as originally conceived?

Reply:
And actually… if you want to read multiple values off of a generator, you can’t force that generator to be restarted every time you call *module*.evaluate.

Reply:
Okay, renamed to *module*.value as discussed.